lib/net.js

```diff
@@ -906,8 +906,11 @@ Socket.prototype.connect = function(options, cb) {
     if (localPort && !util.isNumber(localPort))
       throw new TypeError('localPort should be a number: ' + localPort);
 
-    if (port <= 0 || port > 65535)
-      throw new RangeError('port should be > 0 and < 65536: ' + port);
+    if (!util.isUndefined(options.port) && options.port != port)
```

Comment:
@cjihrig ah, my checks are to see if they _are_. so instead it would be:
```javascript
if (!(options.port >>> 0 === options.port && options.port >= 0))
```